### DISCUSSION TITLE
Athom touch switches experiences....

### DISCUSSION THREAD

**Discussion by btsimonh** [arendst/Tasmota#14907] [Q&A] [answered]
More a discussion than a question...  The below is my experience of Athom pre-flashed with TAS.
If you have experience with Athom, then feel free to add yours...

I recently bought some Athom EU touch wall switches:
2 x 4 way (DS-101JL)
1 x 3 way (SW113EU)
2 x dimmer (SD01-EU-TAS)

They came with TAS 8.5 (marked athom?), and I did upgrade to 10 before REALLY looking hard at the functionality.
(upgrade did seem unexpectedly difficult).

Experiences:
4-way:
The 4 way switches I received worked (for a while) on the bench.  But when installed in the wall, the touch switches just did not want to work.  It turns out they know they had a bad batch....
Basically, these switches have a tuya-style MCU controller for the touch parts - it's not connected to ESP serial, but does have it's own TX/RX pins, so I assume it's an MCU (unmarked 16 pin chip).
My assumption is that the touch calibration is compromised, and it did not like either being in proximity to another, or did not like the earthed metal box in the wall...
Athom replaced them with 3-way on my request (not TOO difficult to deal with, and they did send the new ones by airmail, and did not mention sending the old ones back - so maybe I can fix them (any info on MCU appreciated), or maybe I just got a pair of 4 way relay controllers?).
Summary: Confirm that you will get working models?  then maybe OK.

3-way:
These work OK.  But not WELL.  When considering mounting in the wall, the plastic is easy to deform, and the front panels MUST fit very well to the rear part for reliable touch operation.  Mine are in a square of four, making them difficult to mount without deforming the plastic enough to make the touch buttons unreliable.  I still have some 'tuning' to do on the mounting (considering shimming 1mm off the wall to ensure the front can sit home completely...).
The advantage if these units is that they have an ESP pin per touch switch, per LED, and per relay (did not check, but they do not 'click' like the 4 way - they may be solid state relays of some kind?).  So, they are much more adaptable than the 4 way where the LEDs and relays are tied together.
Summary: ok if mounted carefully.

Dimmer:
They work.  But not WELL in TAS.
No touch problems (apart from the mounting issues discussed above).
I have two main issues. 
1/ the two that I received seem to have different tuya MCU firmware.  One has the central LED lit, the other has it unlit... so they look different.
2/ The tuya MCU and TAS don't talk nice.  The TAS interface does not provide a linear dim (it's all in the top few percent).  The tuya MCU ONLY works after a power-off reboot.  If you restart TAS, the unit does not work.  The tuya MCU seems to send a message about wifi reset continually, which TAS does not respond to (examples or response are in the tuya docs - I will at some point add this to TAS).  I am convinced that there should be a command to reset the tuya MCU, which we could send at TAS restart, but without knowledge of the MCU software implementation, it's a little difficult to know where to start.
(I have to admit that this is my first device with Tuya MCU, and I've not gone deep in TAS docs or source to see if the above can/are solved...).
The touch functionality of dimming and brightening a lamp seems to work (I'm using a cheap LED dimmable, and I had hopes it would be better, but for the moment, I blame the LED lamp; we will see).
Summary: Needs better TAS support, and mount carefully.

Overall summary:
I would not go with touch switches again.
And I would not go with anything that contained/relied on a tuya style MCU unless the manufacturer published at least their protocol implementation, and had provisioned for firmware updates.

I hear Martin Jerry will put out some EU pre-flashed with TAS later in the year... I wish I had waited.

**Comment by ascillato** [comment, marked as answer]
Hi,

I'm sorry to hear that. For issues related to Athom, you should address those to directly them. Tasmota is not related to any manufacturer, hence we can't do anything about your hardware issues.

About your overall summary, my personal opinion is exactly the same as yours, but I'm speaking just for myself on this matter. If I have to give a recommendation I would say the same.

**Reply by barbudor** [reply]
I haven't used any Athom ones but a few Esooli and MiniTiger from Ali
Finally I gave up because of the poor WAF (as well as myself too)
We prefer swithes that "click" with a good mechanical feedback
So finally just moved to a few Shelly 1L (no-neutral) to keep our goo' ol' switches.

**Comment by btsimonh** [comment]
@barbudor - yes, kind of destroyed the WAF factor....
I think I got there now - pending the 'mounting' tweaking....
@ascillato - this was not a comment on Tas's Athom support, I just hoped it would help others in their purchasing decision :)  - so no need to be sorry, and I'm as much responsible for tas a you, so don't apologise! (I contribute where I can :) ).

**Comment by spikerguy** [comment]
Hello All,
I would like to add to this.
![image](https://user-images.githubusercontent.com/48839029/156595642-4eb04e02-b25c-40be-bcae-1db05aae0a4c.png)

ATHOM Touch Switch (DS-101) only worked for 1 day max. It was used on LED only and nothing heavy to kill it. 
ATHOM Smart plug with Power Monitor (TP27Y) lasted only a month or so. 3 out of 4  died within a month. Asked Athom support on Aliexpress to advice they said this.
![image](https://user-images.githubusercontent.com/48839029/156596426-c1b0257d-0382-4519-9c05-73b17215b6e3.png)
From their reply about failure rate, it seems they knew the failure rate already which must be high in normal sense but they might find it low/average as per their expectation.

Taking the software for free and asking their customer to get free support from the same oss project, Well didn't expect this from Athom.
@arendst I respect what you guys at tasmota are doing and I am an OSS consultant, I advice you to ask customers not to buy Athom devices as they have very poor hardware quality, it is just spoiling tasmota's image while increasing the burden of support tasmota community have to provide. 

I would also request Athom company to pay Tasmota for per device sold with tasmota firmware, without any warranty/guarantee of support from tasmota. 

I have come to a conclusion that ATHOM products are not worth the price we pay for tasmota pre-installed then the amount of time wasted on troubleshooting, Instead should have bought a devices which is documented on templates and flash tasmota manually, would have saved some amount.

**Reply by btsimonh** [reply]
I await my failures :).  
The "4 way working for 1 day" could be the same issue I had - I got a response from Athom recently that my touch issue was 'interference', and only solved by a change of touch chip.  Maybe you got the same batch.  They did replace mine with 3way, and quickly.  the 2 x 4-way I have work fine as 4-way relays - for which I have no use but can't bring myself to bin.

Ref "Instead should have bought a devices which is documented on templates" - I had to do my research very quickly on what to buy, and regret I did not spend more time as to where to invest my £120.  
Just because there is a [template](https://templates.blakadder.com/athom_SW11-TAS-4EU.html) on blakadder, does not/cannot provide any indication to quality - it really is a lottery as to if you'll get a device with ESP, and what manufacturing quality to expect.  Tbh, the Athom reviews I found were not bad, but hard to find and confirm.  But next time I would work off personal recommendation only (e.g. martinjerry - but little european stuff yet).

To give Athom their due, at least they DO ship with TAS pre-installed (so you *know* it's esp), and *did* support my buggy hardware.  
You can't guarantee ESP if you order something *hoping* it's esp (e.g. [Calex LED strip](https://templates.blakadder.com/calex-429240.html) - developed our own firmware for the Beken chip rather than bin it).

**Reply by Jimbo1281** [reply]
I have the same with 7 wall touch switches from Athom. The board goes off or fails completely. no way to reset. lights don't work after a while like on kingarts. Also super annoying is the fact that they don't play nice with most of my LED lights in the house.

Just replaced all with SONOFF T0EU and T2EU. Everything works straight away after flashing tasmota and no buzzing sound on any of the LEDS.

The only thing that works constantly and is reliable, is the Athom power Plugs I use for my floor lights.

**Comment by athom-tech** [comment]
As @btsimonh  said, the 4Gang European touch switch does have the problem of touch chip interference. For this reason, we have also contacted the factory and solved it. For those devices that have hardware errors, please take the initiative to contact customer service and we will fix it for you. Issue new equipment. For other equipment problems, please contact customer service to solve

**Reply by spikerguy** [reply]
I did contact the Customer Support and they said it have nothing to do with hardware issue. It does not just have touch issue it have wrong template assigned as default, which is not properly informed to the users.
While having no wifi connection to the router the device would keep powering on/off the switch4 on loop and then they say its a normal function ?
That is a standard function of sonoff basic and they should have made default template as per their device. Atleast not a practical solution for home automation as pre-installed tasmota.
This behavior is officially confirmed by the customer support as default is on sonoff basic. While not advising which template to be used for their device.
Screenshot below.
![image](https://user-images.githubusercontent.com/48839029/156880923-4fd0cad9-8a85-443a-8bfc-e760a1d0768f.png)

Athom Customer service on Aliexpress messaging did not even care to reply to my text on 3/4 Smart Plug with Power Monitor dead issue.
Screeshot below.
![image](https://user-images.githubusercontent.com/48839029/156880891-666f0c92-6e14-445a-8aef-719fff4ffb0c.png)

**Reply by athom-tech** [reply]
Hello spikerguy, as mentioned in the chat log, Tasmota's default configuration module is "sonoff basic", and we did make a template for our hardware equipment. When it leaves the factory, the reason why your switch appears switch4 on/off is that you upgraded version of tasmota and reset the device. For your smart plug without power monitoring, you should check whether the GPIO configuration information of the plug is wrong. If there is any problem, please contact customer service to reissue a new one.

**Reply by spikerguy** [reply]
Hello, 
As shown in the screenshot.  Customer service did not result to my messages. 

Smart plug with power monitor is completed dead when plugged into wall power they were working fine until it stopped, I opened one yesterday and tested whether the pcb is receiving 240v and it seem to reach fine till the pcb so to me it looks like a component must have failed.

For touch switch all 3 are old version and had touch issue since the beginning but I ignored it thinking that I want to use it mostly from the Home assistant instead of manual touch.

I will message customer service againwith reference to this thread.  

Let's see what they have to say. 

Thanks.

**Reply by spikerguy** [reply]
@athom-tech I texted the customer support via aliexpress message and no one replied. So yea its the same like before. 
Hope I get some answers on there.
Thanks.

**Reply by athom-tech** [reply]
@spikerguy Please contact support@athom.tech via email, sometimes AliExpress customer service cannot see the information

**Comment by btsimonh** [comment]
@athom-tech - 
Dimmers are being a problem.
If they are controlled too quickly (?) from TAS, the MCU stops operating the light, requiring a complete power-off to resolve.
Do you have a complete protocol description for the MCU - e.g. is there any way to reset it without a power off?
If we can't resolve this in software, do you have a hardware resolution?

**Reply by btsimonh** [reply]
@athom-tech - @aiden -
On my dimmer issue.  It seems a little worse than the device stopping operating and recovering only on hardware (power) restart.
It seems that even on a power-off restart TAS can send MCU messages in different sequences, and certain sequences render the MCU inoperable.
This is certainly a problem with the TAS Tuya MCU protocol implementation, which I would be more than willing to help with in order to resolve the issue.
You should have my email from correspondence with your support dept.  Please let me know if you do not.  Email me?

best regards,
Simon

**Comment by btsimonh** [comment]
update - Athom are sending me two 3-way switches to temporarily replace my problematic dimmers, so that I can remove them from the wall and look into the Tuya driver in TAS. - thankyou @athom-tech for the very fast response to my email to support.  See tuya related discussion, and note that I'd be very happy if Athom could do some dimmer testing as we progress.  The main issues are around booting and control from TAS (e.g. by Alexa/HA), which seems to kill the MCU's ability to operate the light, even though everything looks fine from the serial protocol.

**Comment by btsimonh** [comment]
Help Wanted with TAS & Tuya testing.  I've reworked the tuya driver, but only have a single device for testing.  I've rasied an issue on my repo here: [issue](https://github.com/btsimonh/Tasmota/issues/11) - where  I've referenced my branch and attached a -lite binary.  If you can help by testing it on other Tuya MCU products, please report/dicsuss there?  I don't want to PR until it's seen some testing other than on my Dimmer.

**Comment by Jason2866** [comment]
@btsimonh Great that you have choosen the Tuya challenge ;-).
Since this discussion is about Athom switch devices (only), your asking for testing is not best placed.
Please open a extra new disscusion, with Tuya testing as topic.
Thx for working on Tasmota.

**Reply by btsimonh** [reply]
I wish I hadn't take the challenge!  It's a pig an has eaten many, many days :).  will start a new discussion, thankyou ...

**Comment by spikerguy** [comment]
@athom-tech was kind enough to send a replacement devices for me.
I see the quality of these devices have improved alot, the 4 gang touch switch some more regulators and also the touch is much better while the issue if device getting reset is still there but thats not an issue from athom, it is just the touch is sensitive and when panel is tightened too much making it bend a little then the glass plate triggers all the touch causing it to reset itself. Once it is reset it goes back to Sonoff Basic and in turn the CH1 starts blinking until we connect it to wifi and set the right template. 
These things have to be documented in detail and publish properly to avoid users breaking the device.
For the smart plugs they sent replacement of UK model which seems to be or higher quality than the one I bought previously.

Thank You Athom for the replacement.

**Reply by spikerguy** [reply]
Update on the smart plug they sent as replacement :

Both the replacement died within 3 months of use, 1 was used on wall panel tablet so it would power on and off multiple times a day so idk if it was too much for the smart plug to handle. 

I am not so keen on investing in Athom products anymore sadly even though they're the only one with tasmota pre-installed. Which made me look at other options like zigbee protocol.

Thank you for the replacement though.
